action.yml

```diff
@@ -0,0 +1,101 @@
+name: "Sourcery GitHub Action"
+
+branding:
+  icon: 'box'
+  color: 'orange'
+
+description: "A GitHub Action for running Sourcery."
+
+inputs:
+  token:
+    description: "The Sourcery token for logging in."
+    required: true
+  target:
+    description: "File(s) or directory(ies) to review."
+    required: false
+    default: "."
+  version:
+    description: "Sourcery CLI version, e.g. '0.12.10'. If empty, default to the latest version."
+    required: false
+    default: ""
+  verbose:
+    description: "Verbose output with explanation and code snippets. Either 'true' or 'false'."
+    required: false
+    default: "true"
+  check:
+    description: "Fail job if issues are found. Either 'true' or 'false'."
+    required: false
+    default: "true"
+  in_place:
+    description: "Change files in place. Either 'true' or 'false'."
+    required: false
+    default: "false"
+  config:
+    description: "Location of the Sourcery YAML configuration file."
+    required: false
+    default: ""
+  diff_ref:
+    description: >
+      Reference to compare the current branch with and run Sourcery on the diff.
+      If empty, run Sourcery on the entire codebase.
+      To review only code that changed in a PR, use "github.event.pull_request.base.sha".
```

Comment:
❔ question: Should `github.event.pull_request.base.sha` be the default value of this? :thinking:

Reply:
Brendan and I debated this a while ago. There are a few points to take into account here:

* this action may be used in many different contexts - not only in PRs, but also as a simple check on pushes to main or to other branches, as well as a periodic check triggered by cron events
* `${{ github.event.pull_request.base.sha }}` as a default only makes sense in PRs, not in the other possible use cases
* `${{ github.event.pull_request.base.sha }}` can only be used if, during the `actions/checkout` action, you pass in `fetch-depth: 0`, otherwise we won't be able to see the full diff
* on the other hand, the "empty" `diff_ref` makes sense in all cases - you just run Sourcery on your entire codebase

Because `github.event.pull_request.base.sha` is not a valid default in all use cases and also requires users to correctly configure their `actions/checkout`, we decided to keep the default as simple as possible.
The "PR" use case has to be documented in our official docs to cover all of its nuances.